### PR TITLE
Removed some unnecessary arguments to avoid warnings

### PR DESCRIPTION
**Pull request by muhammad-tanvir-1211**
Removed some unnecessary arguments in the code to avoid warnings at compile time.

### CODE REVIEW

File: src/operations/blas2/gemv.hpp [hunk 1]
```diff
@@ -185,7 +185,7 @@ SYCL_BLAS_INLINE
                                 cl::sycl::nd_item<1> ndItem) {
   const index_t local_id = ndItem.get_local_id(0);
   const index_t group_id = ndItem.get_group(0);
-  const index_t group_range = ndItem.get_group_range(0);
+  // const index_t group_range = ndItem.get_group_range(0);
```
Comment (mehdi-goli) [inline src/operations/blas2/gemv.hpp:188]: Remove the commented code

File: src/operations/blas3/gemm_load_store.hpp [hunk 2]
```diff
@@ -36,19 +36,19 @@ used no matter what value is passed here.
 * @tparam value_t The type of the matrix data (typically float or double, if
 supported).
 */
-template <size_t vector_size, typename value_t, typename index_t>
```
Comment (mehdi-goli) [inline src/operations/blas3/gemm_load_store.hpp:39]: Rewrite the template so that the index_t will be at the beginning and then you can re-use it for the vector_size. This can be on a separate PR maybe as it requires other files calling it to be modified.

File: src/operations/blas3/gemm_load_store.hpp [hunk 3]
```diff
@@ -36,19 +36,19 @@ used no matter what value is passed here.
...
+template <int vector_size, typename value_t, typename index_t>
 struct Packetize {
 #ifdef GEMM_VECTORIZATION_SUPPORT
   using PacketType = cl::sycl::vec<value_t, vector_size>;
-  static constexpr size_t packet_size = vector_size;
+  static constexpr int packet_size = vector_size;
```
Comment (mehdi-goli) [inline src/operations/blas3/gemm_load_store.hpp:43]: use `index_t` after the above change

File: src/operations/blas3/gemm_load_store.hpp [hunk 4]
```diff
@@ -36,19 +36,19 @@ used no matter what value is passed here.
...
   template <index_t dimension>
   SYCL_BLAS_INLINE static constexpr bool check_size() {
     return packet_size == 1 || dimension == packet_size;
   }
 #else
   // In the case where vectorization is not enabled, always set to 1
   using PacketType = cl::sycl::vec<value_t, 1>;
-  static constexpr size_t packet_size = 1;
+  static constexpr int packet_size = 1;
```
Comment (mehdi-goli) [inline src/operations/blas3/gemm_load_store.hpp:51]: use `index_t` after the above change

File: src/operations/blas3/gemm_local.hpp [hunk 5]
```diff
@@ -254,7 +254,7 @@ class Gemm<input_t, output_t, DoubleBuffer, NbcA, NbcB, ClSize, TileType,
         id.get_group_range(0) / get_workgroup_cluster();
 
     auto scratch = scratch_acc.localAcc.get_pointer();
-    using ScratchPointerType = decltype(scratch);
+    // using ScratchPointerType = decltype(scratch);
```
Comment (mehdi-goli) [inline src/operations/blas3/gemm_local.hpp:257]: remove the commented code

File: src/interface/blas3/backend/intel_gpu.hpp [hunk 6]
```diff
@@ -39,8 +39,8 @@ typename executor_t::policy_t::event_t _gemm(
     gemm_batch_type_t batch_type) {
   if (batch_type == gemm_batch_type_t::interleaved) {
     return blas::Gemm_Launcher<
-        64, false, false, false, 64, Tile<4, 4, 4, 4, 1, 1, 1, 1, 4, 4>, _t_a, _t_b,
-        static_cast<int>(gemm_memory_t::no_local),
+        64, false, false, false, 64, Tile<4, 4, 4, 4, 1, 1, 1, 1, 4, 4>, _t_a,
```
Comment (Bensuo) [inline src/interface/blas3/backend/intel_gpu.hpp:42]: Why has this formatting changed? The clang-format file hasn't been updated as far as I can see.

File: src/operations/blas1_trees.hpp [hunk 7]
```diff
@@ -84,7 +84,7 @@ struct DetectScalar<cl::sycl::half> {
   using element_t = cl::sycl::half;
   static element_t get_scalar(element_t &scalar) { return scalar; }
 };
-#endif // BLAS_DATA_TYPE_HALF
+#endif  // BLAS_DATA_TYPE_HALF
```
Comment (Bensuo) [inline src/operations/blas1_trees.hpp:87]: Another strange formatting change.

**Review by Bensuo — APPROVED** [review]
LGTM, though I am confused by some of the strange formatting changes. Was clang-format run on this?

**Review by ProGTX — APPROVED** [review]
